### PR TITLE
fix: muted by incoming call while in call SQCALL-475

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQCALL-475" title="SQCALL-475" target="_blank">BugSQCALL-475</a>  iOS: User gets muted in a group call automatically on receiving a new call and hanging up the new call 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

when receiving an incoming call while already in a call, the user is automatically muted in the ongoing call.

### Causes (Optional)

we don't check that there is no active call when muting 

### Solutions

check that there is no active call before muting

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Tests/Source/Calling/WireCallCenterV3Tests.swift [hunk 1]
```diff
@@ -849,15 +849,47 @@ class WireCallCenterV3Tests: MessagingTest {
...
+        let incomingCall = CallSnapshotTestFixture.callSnapshot(
+            conversationId: incomingCallConversationId,
+            callCenter: sut,
+            clients: [],
+            state: incomingState
+        )
+
+        sut.callSnapshots = [activeCallConversationId: activeCall,
+                           incomingCallConversationId: incomingCall]
+        sut.muted = false
```
Comment (KaterinaWire) [inline Tests/Source/Calling/WireCallCenterV3Tests.swift:885]: If this test checks that it does not mute the microphone, so the initial state must be true. Am I wrong?

Comment (David-Henner) [reply Tests/Source/Calling/WireCallCenterV3Tests.swift:885]: I don't think so, we want to verify that `muted` stays `false`

Comment (sb88k) [reply Tests/Source/Calling/WireCallCenterV3Tests.swift:885]: Then, maybe we can have different test case name like
`testThatTheMicrophoneStaysMuted_WhenHandlingIncomingGroupCall_WhileAlreadyInCall`

Comment (sb88k) [reply Tests/Source/Calling/WireCallCenterV3Tests.swift:885]: Ah, you are right.  Now I fully got what this test is trying to check. 👍

**Review by sb88k — APPROVED** [review]

File: Tests/Source/Calling/WireCallCenterV3Tests.swift (hunk 1 above)
Comment (KaterinaWire) [reply Tests/Source/Calling/WireCallCenterV3Tests.swift:885]: @David-Henner Thanks for the explanation! Now it's clear to me, you're right.

**Review by KaterinaWire — APPROVED** [review]

**Comment by David-Henner** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @David-Henner! Merging PR for command **release patch** for PR #**1668**. Will create a **patch** release after merged.
Next version is 412.3.6.

**Pull request merged**